cs3api4lab/api/cs3apismanager.py

```diff
@@ -422,15 +416,29 @@ def _save_directory(self, path):
 
     @asyncify
     def _check_write_permissions(self, path):
-
         parent = self._get_parent_path(path)
-        stat = self.file_api.stat_info(parent)
-        if not ShareUtils.map_permissions_to_role(stat['permissions']) == 'editor':
+        file_info = self.file_api.stat_info(parent)
+
+        if not self._is_editor(file_info):
             raise web.HTTPError(403, u'The path %s is not writable' % parent)
-        # check if the path is a received share with editor permissions
-        for share in self.share_api.list_received()['content']:
-            if share['path'] == path and share['writable'] == False:
-                raise web.HTTPError(403, u'The share %s is not writable' % path)
+
+    def _is_editor(self, file_info):
+        is_editor = False
+
+        if not file_info:
+            return is_editor
+
+        # check file permissions
+        if ShareUtils.map_permissions_to_role(file_info['permissions']) == 'editor':
```

Comment:
This is returning an enum, so we should be comparing with `Role.EDITOR` and not its string (same below).

Reply:
@diocas It's fixed, right now I use Role.Editor